src/app/shared/services/error.service.ts

```diff
@@ -83,11 +83,21 @@ export class ErrorService {
                         break;
 
                     case AuthError.INVALID_JWT:
+                        if (!errorRes.url.includes('/api/user/token/refresh')) {
+                            action$ = this.takeAction(() => this.userService.refreshUserToken().pipe(
+                                this.handleError,
+                                concatMap(() => _caught)
+                            ));
+                            break;
+                        }
+                    // tslint:disable-next-line:no-switch-case-fall-through
                     case AuthError.MISSING_JWT:
                     case AuthError.CANNOT_REGISTER_JWT:
                     case UserError.USER_JWT_MISSING:
-                        action$ = this.takeAction(() => of(this.userService.logout())
-                            .pipe(this.router.navigate['/']));
+                    case undefined: // TODO: backend returns 400 with no err_code. fix this. this is workaround.
```

Comment:
@githinho,
When I try to get a new access token with refresh token route (POST `/api/user/token/refresh`), I get HTTP error 400, but the `err_code` field is empty. Can I get some code here to be consistent with the rest of the responses?

Reply:
The code is 0208: https://github.com/AMPnet/docs/blob/master/ErrorCodes.md#authentication---prefix-02